File: tests/test_data/multi_language/Calculator.scala
// Scala test file
package demo

trait MathOperations {
  def calculate(a: Int, b: Int): Int
}

class Calculator(val name: String) extends MathOperations {
  def calculate(a: Int, b: Int): Int = a + b

  def add(a: Int, b: Int): Int = a + b

  val version: String = "1.0"
}

object Calculator {
  def create(): Calculator = new Calculator("default")
}

case class Point(x: Double, y: Double)

sealed trait Shape

File: tests/test_data/multi_language/Calculator.php
<?php
namespace App;

class Calculator {
    private float $result;

    public function __construct() {
        $this->result = 0;
    }

    public function add(float $value): self {
        $this->result += $value;
        return $this;
    }

    public static function create(): self {
        return new self();
    }
}

interface MathOperations {
    public function calculate(float $a, float $b): float;
}

trait Logging {
    public function log(string $msg): void {
        echo $msg;
    }
}

function standalone_add(float $a, float $b): float {
    return $a + $b;
}
